packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```diff
@@ -920,6 +921,43 @@ export class ComponentDecoratorHandler implements
     return compileResults(fac, def, classMetadata, 'ɵcmp');
   }
 
+  private _transformDecoratorToInlineResources(
+      dec: Decorator, component: Map<string, ts.Expression>, styles: string[],
+      template: ParsedTemplateWithSource): Decorator {
+    if (dec.name !== 'Component') {
+      return dec;
+    }
+
+    // If no external resources are referenced, preserve the original decorator
+    // for the best source map experience when the decorator is emitted in TS.
+    if (!component.has('templateUrl') && !component.has('styleUrls')) {
+      return dec;
+    }
+
+    const metadata = new Map(component);
+
+    // Set the `template` property if the `templateUrl` property is set.
+    if (metadata.has('templateUrl')) {
+      metadata.delete('templateUrl');
+      metadata.set('template', ts.createStringLiteral(template.content));
+    }
+
+    // Set the `styles` property if the `styleUrls` property is set.
+    if (metadata.has('styleUrls')) {
+      metadata.delete('styleUrls');
+      metadata.set('styles', ts.createArrayLiteral(styles.map(s => ts.createStringLiteral(s))));
+    }
+
+    // Convert the metadata to TypeScript AST object literal element nodes.
+    const newMetadataFields: ts.ObjectLiteralElementLike[] = [];
+    for (const [name, value] of metadata.entries()) {
+      newMetadataFields.push(ts.createPropertyAssignment(ts.createStringLiteral(name), value));
```

Comment:
I think that by using `ts.createStringLiteral(name)` you end up with objects like:

```ts
{
  "template": ...
}
```

rather than 

```ts
{
  template: ...
}
```

So I think that this should just be:

```suggestion
      newMetadataFields.push(ts.createPropertyAssignment(name, value));
```

Reply:
I intentionally used the string-literal key variant at first because technically the key could also be some arbitrary property containing a dash. This is obviously not allowed as per our `@Component` decorator, but folks could do it using e.g. `@ts-ignore`. Regardless though.. I'm updating it to the `ts.Identifier` variant because this seems really like an edge-case..